mcs/class/System/Mono.Net.Security/AsyncProtocolRequest.cs

```diff
@@ -70,7 +70,10 @@ public void Reset ()
 		{
 			Offset = Size = 0;
 			TotalBytes = 0;
-			Buffer = new byte[InitialSize];
+			if (Buffer.Length <= InitialSize)
+			    Array.Clear (Buffer, 0, Buffer.Length);
```

Comment:
Maybe I should be more strict and only do it if its the exact size as the buffer?